sql/core/src/main/scala/org/apache/spark/sql/functions.scala

```diff
@@ -737,6 +735,21 @@ object functions {
   }
 
   /**
+   * Creates a new struct column with given field names and columns.
+   * The input columns should be of length 2*n and follow (name1, col1, name2, col2),
+   * name* should be String Literal
+   *
+   * @group normal_funcs
+   * @since 1.5.0
+   */
+  @scala.annotation.varargs
+  def named_struct(cols: Column*): Column = {
+    require(cols.length % 2 == 0,
+      s"named_struct expects an even number of arguments.")
+    CreateNamedStruct(cols.map(_.expr))
+  }
+
+  /**
```

Comment:
I found a little difficult here to name the parameters in this API, since it should be fieldName1, value1, fieldName2, value2, I'll consider this again.